rgdps/models/user_relationship.py

```diff
@@ -0,0 +1,45 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from datetime import datetime
+from typing import Any
+from typing import Mapping
+
+from rgdps.constants.users import UserRelationshipType
+
+
+@dataclass
+class UserRelationship:
+    id: int
+    relationship_type: UserRelationshipType
+    user1_id: int
+    user2_id: int
+    post_ts: datetime
+    seen_ts: datetime | None
+
+    @staticmethod
+    def from_mapping(relationship_dict: Mapping[str, Any]) -> UserRelationship:
```

Comment:
nit: param should prob be called `mapping` (ik this isnt consistent in the codebase but it will be in a pr sooner or later)

Reply:
Changed in 4e22e55